### DISCUSSION TITLE
More  brightness

### DISCUSSION THREAD

**Discussion by chrisdal2000** [mrcodetastic/ESP32-HUB75-MatrixPanel-DMA#559] [General]
Hi to all,
i have a 64x32 led
![viber_image_2024-01-08_23-08-25-057](https://github.com/mrfaptastic/ESP32-HUB75-MatrixPanel-DMA/assets/140742911/b8d07e32-ee98-445c-a13f-67be67e2ed74)

and with that code

void setup() {
  //                             R1, G1, B1,           R2,   G2, B2,             A,  B,  C,  D,    E,  LAT, OE,CLK, 
  HUB75_I2S_CFG::i2s_pins _pins={25, 26, 27,           14,   12, 13,            23, 19,  5, 17,   0,   32, 33, 15,};
  HUB75_I2S_CFG mxconfig(
    PANEL_RES_X, // Module width
    PANEL_RES_Y, // Module height
    PANEL_CHAIN, // chain length
    _pins, HUB75_I2S_CFG::FM6126A     //FM6126A           DP3246_SM5368
  );
  
  mxconfig.double_buff=true;
  mxconfig.clkphase = false;
  mxconfig.min_refresh_rate=60;
  mxconfig.i2sspeed=HUB75_I2S_CFG::HZ_20M;  

  // Display Setup
  dma_display = new MatrixPanel_I2S_DMA(mxconfig);
  dma_display->begin();
  dma_display->setPanelBrightness(255);
  dma_display->clearScreen();
  dma_display->fillScreen(myBLACK); 

  dma_display->setRotation(0);  
  dma_display->flipDMABuffer();
  //setCpuFrequencyMhz(240);
}

int xx=0, yy, dx;
void loop() {
  yy=0; dx=1;
  for(int x2=0;x2<=xx;x2=x2+1){          
    for(int y=0;y<=yy;y=y+1){      
        if (dx>0)
        {dma_display->drawPixel(x2,y, myBLUE );}
        else
        {dma_display->drawPixel(x2,y, myRED );}
    }
    yy=yy+dx;
    if (yy==32) {dx=-1;}
    if (yy==0) {dx=1;}
  }
  dma_display->flipDMABuffer();  
  delay(500);
  dma_display->clearScreen();
  xx=xx+1;
  if (xx==64) {xx=0;}
}

show that
![viber_image_2024-01-08_22-59-08-669](https://github.com/mrfaptastic/ESP32-HUB75-MatrixPanel-DMA/assets/140742911/04b7b1f1-177b-4061-89cb-dc77327846c1)
All is ok and the brightness is good (for internal house). 
But when change the driver to  HUB75_I2S_CFG::DP3246_SM5368
then take that

![viber_image_2024-01-08_22-59-07-894](https://github.com/mrfaptastic/ESP32-HUB75-MatrixPanel-DMA/assets/140742911/32e84ce7-4ddc-43de-870a-25a29e3d05ca)
The image it is not same (has problem) but the brightness at middle is very very bright.
Here is and the video for 1st and 2nd code.

https://github.com/mrfaptastic/ESP32-HUB75-MatrixPanel-DMA/assets/140742911/cba17236-a450-4553-bdd9-e8630f61e764

https://github.com/mrfaptastic/ESP32-HUB75-MatrixPanel-DMA/assets/140742911/5e5c8a64-b107-4237-939d-af8955a31df6

Can i have that brightness with FM6126A or make the DP3246_SM5368 to work ok with that brightness?

I see the code here (ESP32-HUB75-MatrixPanel-I2S-DMA.cpp)
else if (m_cfg.driver == HUB75_I2S_CFG::DP3246_SM5368) 
      {
        row[ESP32_TX_FIFO_POSITION_ADJUST(x_pixel)] = 0x0000;
      }
else
why show all data at row 0.

Sorry for the english.

**Comment by board707** [comment]
> Can i have that brightness with FM6126A or make the DP3246_SM5368 to work ok with that brightness?

Shortly - no.
When the image is displayed on the entire matrix, the total brightness is divided into all rows and therefore the brightness of each pixel is less. In the second case, the pixels of all lines is displayed on a single line, so it looks brighter, but all other lines do not light up at all

**Reply by chrisdal2000** [reply]
thanks for the reply.

This means the hardware is ok and the led can be more brightness from normal. The problem is the software.
There are a other way to have more brightness?

**Reply by board707** [reply]
> The problem is the software. There are a other way to have more brightness?

No, there is no problem with the software. This is how the panels works - they show the image by quickly switching between lines. That means each line is lit only 1/T part of the total time, so the visible brightness of each line is T times less than the maximum brightness of an individual diode.
There is no way to light every led on panel with maximum brightness at the same time.

**Comment by board707** [comment]
By the way,  that the greater the number of scans in the matrix - for example, 16s for 64x32 16 scans - the lower its brightness. One of the reasons why manufacturers produce matrices with multi-line scanning - for example, the same 64x32, but with scan 8s - is to increase their brightness.
Another thing from this - one matrix 64x64 32s has brightness only a half relatively to two 64x32 16s panels (with the same leds and drivers) .
